UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/ReactionManager.cs

```diff
@@ -191,50 +192,58 @@ private void UpdateMe()
 
 		private void ProcessWindNodes(MetaDataNode windyNode)
 		{
+			if (windyNode.WindForce < MINIMUM_WIND_FORCE) return;
 			windyNode.WindData[(int) PushType.Wind] = (Vector2) windyNode.WindDirection * (windyNode.WindForce);
 
-			var registerTiles = matrix.GetRegisterTile(windyNode.LocalPosition, true);
-			for (int i = 0; i < registerTiles.Count; i++)
+			Span<RegisterTile> registerTiles = matrix.GetRegisterTile(windyNode.LocalPosition, true).ToArray();
+
+			CheckForPushables(registerTiles, windyNode);
+
+			windyNode.WindForce = (windyNode.WindForce * ((RollingAverageN - 1) / RollingAverageN));
+			if (windyNode.WindForce < MINIMUM_WIND_FORCE)
+			{
+				winds.Remove(windyNode);
+				windyNode.WindForce = 0;
+				windyNode.WindDirection = Vector2Int.zero;
+				windyNode.WindData[(int) PushType.Wind] = Vector2.zero;
+				return;
+			}
+
+			if(activeWindEffectSpots.Contains(windyNode)) return;
+			windyNode.windEffectTime = WindParticleBlockTime;
+			activeWindEffectSpots.Add(windyNode);
+
+			windEffectNodes.Add(new WindEffectData(windyNode.PositionMatrix.NetworkedMatrix.MatrixSync.netId,
+				windyNode.LocalPosition, windyNode.WindDirection));
+		}
+
+		private void CheckForPushables(Span<RegisterTile> registerTiles, MetaDataNode windyNode)
```

Comment:
since this returns void and it seems it simply acts on what it finds, then I think this should be called something like ``PushPushables`` instead of check. Check kind of implies it will return a boolean and we can call it to test something.

Reply:
I changed the name 👍